### PR TITLE
Added tests for MP-JWT subsystem

### PR DESCRIPTION
**Pull request by honza-kasik**
Added test for subsystem add/removal

Please make sure your PR meets the following requirements:
- [x] Pull Request contains a description of the changes
- [x] Pull Request does not include fixes for multiple issues/topics
- [x] Code is formatted, imports ordered, code compiles and tests are passing (not right now, #132 needs to be merged and this needs to be rebased on top of it)
- [ ] N/A Link to the passing job is provided
- [x] Code is self-descriptive and/or documented
- [x] Description of the tests scenarios is included (see #46)

### CODE REVIEW

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/MicroProfileJwtSubsystemTestCase.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    @Before
+    public void after() throws IOException, TimeoutException, InterruptedException {
+        final Operations ops = new Operations(client);
+
+        if (wasSubsystemRemovedInPrepare) {
+            ops.add(MP_JWT_SUBSYSTEM_ADDRESS).assertSuccess();
+            new Administration(client).reloadIfRequired();
+        }
+
+        if (client != null) {
```
Comment (rsvoboda) [inline microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/MicroProfileJwtSubsystemTestCase.java]: Is this check really needed ?

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/MicroProfileJwtSubsystemTestCase.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        if (ops.exists(MP_JWT_SUBSYSTEM_ADDRESS)) {
+            ops.remove(MP_JWT_SUBSYSTEM_ADDRESS).assertSuccess();
+            new Administration(client).reloadIfRequired();
+            wasSubsystemRemovedInPrepare = true;
+        }
+    }
+
+    @Before
+    public void after() throws IOException, TimeoutException, InterruptedException {
+        final Operations ops = new Operations(client);
```
Comment (rsvoboda) [inline microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/MicroProfileJwtSubsystemTestCase.java]: Can be moved to `if (wasSubsystemRemovedInPrepare) ...`

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/MicroProfileJwtSubsystemTestCase.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        final Administration administration = new Administration(client);
+
+        ops.add(MP_JWT_SUBSYSTEM_ADDRESS).assertSuccess();
+        administration.reloadIfRequired();
+        Assert.assertTrue(ops.exists(MP_JWT_SUBSYSTEM_ADDRESS));
+        ops.remove(MP_JWT_SUBSYSTEM_ADDRESS).assertSuccess();
+        administration.reloadIfRequired();
+        Assert.assertFalse(ops.exists(MP_JWT_SUBSYSTEM_ADDRESS));
+    }
+
```
Comment (rsvoboda) [inline microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/MicroProfileJwtSubsystemTestCase.java]: What about calling wasSubsystemRemovedInPrepare twice ? Second call should fail + error message should be reasonable.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/MicroProfileJwtSubsystemTestCase.java (hunk 1 above)
Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/MicroProfileJwtSubsystemTestCase.java]: Sorry, old habits (this check was required from me when working on HAL testsuite). Removed.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/MicroProfileJwtSubsystemTestCase.java (hunk 2 above)
Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/MicroProfileJwtSubsystemTestCase.java]: Done, thanks.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/MicroProfileJwtSubsystemTestCase.java (hunk 3 above)
Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/MicroProfileJwtSubsystemTestCase.java]: I'm sorry, I'm not sure I follow - are you pointing out that I should try to remove the subsystem for second time and verify, the operation fails?

Comment (rsvoboda) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/MicroProfileJwtSubsystemTestCase.java]: yes, new test case candidate.

Comment (rsvoboda) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/MicroProfileJwtSubsystemTestCase.java]: this can be new PR ...

**Comment by rsvoboda** [comment]
[ERROR] Failed to execute goal net.revelc.code:impsort-maven-plugin:1.3.2:check (default-cli) on project microprofile-jwt: Imports are not sorted in /home/travis/build/jboss-eap-qe/eap-microprofile-test-suite/microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/MicroProfileJwtSubsystemTestCase.java -> [Help 1]

please fix it

**Comment by honza-kasik** [comment]
Done

**Pull request merged**